### ISSUE TITLE
Standardize `ion_set` indexing and refine `Set` structure in the DSPMDE property package

### ISSUE THREAD

**Issue by lbibl** [watertap-org/watertap#695]
Currently, in DSPMDE properties possessed only by ions are indexed by `ion_set | solute_set`.  The chem team agreed to standardize this to let ion properties indexed by`ion_set` only.  This is on a clear to do list. 

Secondly, the chem team discussed the potential refinement of the overall `Set` structure in DSPMDE.  The current version has `solute_set` being nonionic species separated from `ion_set`.  A better solution is to have `cation_set` and `anion_set`  IN `ion_set`; `ion_set` and `nonion_set`(or `molecule/neutral_set` to consider) IN `solute_set`; `solute_set` and `"H2O"` (`solvent_set`) IN `component_list`.   This can be done by directly declaring everything as `Set( )` and grouping an input dictionary into them by a loop in DSPMDE.  However, there are some considerations about being aligned with and whether to use the IDAES convention, on which side there are also some possible changes in the future.  The decision and action on this are thus on hold and tracked by this issue. 

TO DO:

- [x] Standardize ion property index ( PR #699 )
- [ ] Decide and act on refining the `Set()` structure

**Comment by adam-a-a** [comment]
@lbibl Thanks for putting this together. Quick comment- I would opt for `neutral_set` with regard to solutes with no charge.

Additionally, we might want to rename this property package something else that doesn't mention "DSPMDE" explicitly. Although this property model started out as a quick creation of mine so that I could get the nanofiltration DSPM-DE model going, the prop model has become a bit more general and is being used by a few other units. A trivial change, but something like `ion_property_model` would seem fine to me. That is, no need to overthink the name.

**Comment by lbibl** [comment]
> @lbibl Thanks for putting this together. Quick comment- I would opt for `neutral_set` with regard to solutes with no charge.
> 
> Additionally, we might want to rename this property package something else that doesn't mention "DSPMDE" explicitly. Although this property model started out as a quick creation of mine so that I could get the nanofiltration DSPM-DE model going, the prop model has become a bit more general and is being used by a few other units. A trivial change, but something like `ion_property_model` would seem fine to me. That is, no need to overthink the name.

I also thought of changing the file name, but I also would like a sense of historic flavor to keep the dspmde name (just to memorize how it was born :) ) .  If renaming, my first idea of it is `solution_property` or even `liquid_solution_property`, for solution is the basic term describing a mixture with one component (solvent) solvating others (solutes).  

This prompted a quick thought related to our"phase indexing" discussion:  `"Liq"` being the only current phase makes me speculate IDAES convention refer `phase` majorly to liquid, vapor, and solid,  (i.e., the most common matter state), rather than finer distinction, e.g., in different liquid phases (mixture of water, oil, etc.)  In this aspect, the significance of indexing phase is about the likelihood to use it on liquid-vapor-solid systems.  Another consideration, which I think is what @andrewlee94 implicated, is that if we index phase and do use it on a multiple-phase system, will the computational methods for a certain property in different (e.g., 2) phases be common? Indexing phase would be more beneficial if the answer is yes (otherwise, Prop[p1] and Prop[p2] will be handled in very different means anyways so indexing them becomes trivial).  In this respect, phase indexing might be more meaningful for those "quantity properties" in our property pack, assuming that they obey homogeneity in the scale of interest in possible phases we'll deal with (water, air, and solid).   But if some property will be computed very differently in a different phase or even not definable, it will be less meaningful to index phase for it.    This para of statement may not necessarily lead to any immediate actions, since neither the current phase indexing causes any trouble nor we have emerging needs to deal with multi-phase systems.  I hope this documents some ideas spurred from our meeting discussion.

**Comment by lbibl** [comment]
The remaining task of this issue and its addressing PR #860 is pending due to an unidentified solving failure with the nanofiltration DSPMDE model.  The failure is inconsistent across a few os platforms (e.g., GitHub action (failed with unknown error), my macOS (passed), @adam-a-a's (failed)  and @bknueven's (failed with a different error from GitHub)  local PC).  Possible causes can be in the indexing of the unit or property model, and/or other scaling/model feasibility issues.   Addressing this isn't urgent for the time being so we may keep this issue and PR #860 pending for a decision or resolution.

**Comment by lbibl** [comment]
This issue will be closed after #860 gets merged.  My other notes were written in #860 and #916.
